packages/autorest.go/src/generator/helpers.ts

```diff
@@ -20,9 +20,16 @@ export async function contentPreamble(session: Session<CodeModel>, packageName?:
     packageName = session.model.language.go!.packageName;
   }
   const headerText = comment(await session.getValue('header-text', 'MISSING LICENSE HEADER'), '// ');
-  let text = `//go:build go1.18\n`;
-  text += `// +build go1.18\n\n${headerText}\n// DO NOT EDIT.\n\n`;
-  text += `package ${packageName}\n\n`;
+  let text = `//go:build go1.18\n// +build go1.18\n\n`
+  // ensure tools recognize the file as generated according to
+  // https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source
+  text += headerText.replace(/^\/\/ Code generated .*\.$/m, '$& DO NOT EDIT.');
+  if (text.match(/^\/\/ Code generated .* DO NOT EDIT\.$/m)) {
+    text += "\n"
+  } else {
+    text += "// Code generated by @autorest/go. DO NOT EDIT.\n\n"
```

Comment:
Is there a case where this branch is executed or is this just defensive?

Reply:
We land here when there's no license header or the header doesn't have a "Code generated ..." line. I see the same MIT header everywhere so practically speaking it's probably just future proofing.

Looking at this again, I saw it could be simplified by moving a newline.